### PR TITLE
fix(voice): generate EncryptionMode instead of using an enum

### LINKED ISSUE #735: Unable to use voice capabilities
Kord version: 0.8.0-M17
Lavaplayer version: walkyst lavaplayer fork v1.3.99.2

Source code is available at [pivo-sound](https://github.com/H4kt/pivo-sound)

Stacktrace:
```
[DefaultDispatcher-worker-5] ERROR dev.kord.voice.gateway.DefaultVoiceGateway - dev.kord.voice.EncryptionMode does not contain element with name 'aead_xchacha20_poly1305_rtpsize' at path $.d.modes[2]
kotlinx.serialization.SerializationException: dev.kord.voice.EncryptionMode does not contain element with name 'aead_xchacha20_poly1305_rtpsize' at path $.d.modes[2]
	at kotlinx.serialization.json.internal.JsonNamesMapKt.getJsonNameIndexOrThrow(JsonNamesMap.kt:63)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeEnum(StreamingJsonDecoder.kt:351)
	at kotlinx.serialization.internal.EnumSerializer.deserialize(Enums.kt:113)
	at kotlinx.serialization.internal.EnumSerializer.deserialize(Enums.kt:79)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:70)
	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableValue(AbstractDecoder.kt:43)
	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableElement(AbstractDecoder.kt:70)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableElement(StreamingJsonDecoder.kt:162)
	at kotlinx.serialization.encoding.CompositeDecoder$DefaultImpls.decodeSerializableElement$default(Decoding.kt:533)
	at kotlinx.serialization.internal.CollectionLikeSerializer.readElement(CollectionSerializers.kt:80)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.readElement$default(CollectionSerializers.kt:51)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.merge(CollectionSerializers.kt:36)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.deserialize(CollectionSerializers.kt:43)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:70)
	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableValue(AbstractDecoder.kt:43)
	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableElement(AbstractDecoder.kt:70)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableElement(StreamingJsonDecoder.kt:162)
	at dev.kord.voice.gateway.Ready$$serializer.deserialize(VoiceEvent.kt:74)
	at dev.kord.voice.gateway.Ready$$serializer.deserialize(VoiceEvent.kt:74)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:70)
	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableValue(AbstractDecoder.kt:43)
	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableElement(AbstractDecoder.kt:70)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableElement(StreamingJsonDecoder.kt:162)
	at kotlinx.serialization.encoding.CompositeDecoder$DefaultImpls.decodeSerializableElement$default(Decoding.kt:533)
	at dev.kord.voice.gateway.VoiceEvent$DeserializationStrategy.deserialize(VoiceEvent.kt:46)
	at dev.kord.voice.gateway.VoiceEvent$DeserializationStrategy.deserialize(VoiceEvent.kt:21)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:70)
	at kotlinx.serialization.json.Json.decodeFromString(Json.kt:95)
	at dev.kord.voice.gateway.DefaultVoiceGateway.read(DefaultVoiceGateway.kt:147)
	at dev.kord.voice.gateway.DefaultVoiceGateway.access$read(DefaultVoiceGateway.kt:47)
	at dev.kord.voice.gateway.DefaultVoiceGateway$readSocket$2.emit(DefaultVoiceGateway.kt:136)
	at dev.kord.voice.gateway.DefaultVoiceGateway$readSocket$2.emit(DefaultVoiceGateway.kt:134)
	at kotlinx.coroutines.flow.FlowKt__ChannelsKt.emitAllImpl$FlowKt__ChannelsKt(Channels.kt:62)
	at kotlinx.coroutines.flow.FlowKt__ChannelsKt.access$emitAllImpl$FlowKt__ChannelsKt(Channels.kt:1)
	at kotlinx.coroutines.flow.FlowKt__ChannelsKt$emitAllImpl$1.invokeSuspend(Channels.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)

```

### PR DESCRIPTION
**Pull request by lost-illusi0n**
tested with both sending and receiving audio.
fixes #735.

### CODE REVIEW

File: voice/src/main/kotlin/EncryptionMode.kt [hunk 1]
```diff
@@ -1,28 +1,15 @@
...
-@Serializable
-public enum class EncryptionMode {
-    @SerialName("xsalsa20_poly1305")
-    XSalsa20Poly1305,
-
-    @SerialName("xsalsa20_poly1305_suffix")
-    XSalsa20Poly1305Suffix,
+@file:GenerateKordEnum(
+    name = "EncryptionMode", valueType = STRING,
+    docUrl = "https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-trigger-types",
```
Comment (Lukellmann) [inline voice/src/main/kotlin/EncryptionMode.kt:3]: this link is wrong, if there is none, just leave it out

File: voice/src/main/kotlin/EncryptionMode.kt [hunk 2]
```diff
@@ -1,28 +1,15 @@
...
+    entries = [
```
Comment (Lukellmann) [inline voice/src/main/kotlin/EncryptionMode.kt:4]: is it intended that some encryption modes are missing that we had before?

File: voice/src/main/kotlin/EncryptionMode.kt (hunk 1 above)
Comment (lost-illusi0n) [reply voice/src/main/kotlin/EncryptionMode.kt:3]: huh weird. i remember changing that from when i copied it. guess i undid it somehow.

File: voice/src/main/kotlin/EncryptionMode.kt (hunk 2 above)
Comment (lost-illusi0n) [reply voice/src/main/kotlin/EncryptionMode.kt:4]: yep, these are the only officially supported audio encryption modes. the others were there just to allow the enum to be deserialized.

File: voice/src/main/kotlin/EncryptionMode.kt (hunk 1 above)
Comment (Lukellmann) [reply voice/src/main/kotlin/EncryptionMode.kt]: Could you also regenerate the result? It still has the wrong link.

**Review by Lukellmann — APPROVED** [review]
Thanks!

**Pull request merged**